Unity-Technologies/Graphics
# DLSS in HDRP - Deep Learning Super Sampling integration

**Integrating NVIDIA's Deep Learning Super Sampling into HDRP**

### Purpose of this PR
* Adding new DLSS upscaler into HDRP.
* Adding the concept of upscaling before post process. 
* Post process pipeline massaging and refactor of destination size. Added the concept of post process size scales, for sampling and rendering, since now the dynamic resolution upscaler could run before or after post processes.
* Adding new UI controls of DLSS
* Adding new tests images for DLSS
* Added new debug panel in the render pipeline analysis pane, to check the internal state of NVIDIA unity plugin.

### Testing status:
To test, just get the latest trunk version, which contains DLSS backend features.

The following steps have been tested locally:

1.  Open the package manager, select in the filter "Built-in" , ensure to enable the "NVIDIA" core native module in your project.
2.  Ensure you dont have render doc on otherwise, the feature wont be detected. In the HDRP quality settings, right under Dynamic Resolution section, enable "tick on the Enable NVIDIA Deep Learning Super Sampling Checkbox"
3.  Ensure your desired camera has Dynamic Resolution Enabled (check in the inspector the checkbox is on)
4.  Read the info box that will appear. This info box will let you know if DLSS is detected or not on this session. If not, then the fallback regular antialiasing is used, and the standard DRS path is executed.
5.  Inspect / play around with software/hardware dx12/dx11/vulkan  APIs of the effect

This effect must not be present in any other platform other than Windows, and in the case any GPU other than NVIDIA's is present, the effect must not be enabled.

If you are measuring performance, please ensure that you are looking at GPU time, using the GPU profiler, and not the CPU profiler. Note too that DLSS wont make your frame run faster unless your scene is highly bound by gpu draw calls.
One example is spaceship demo.

### Known Testing Issues:
We have 1 known crash, vulkan on intel CPU is now crashing %100 of the time, this is being addressed in a separate PR:
https://ono.unity3d.com/unity/unity/pull-request/125566/_/graphics/fix-nvapi-vulkan-crash
